src/Microsoft.ML.Transforms/MutualInformationFeatureSelectingTransformer.cs

```diff
@@ -56,87 +61,136 @@ public sealed class Arguments : TransformInputBase
             public int NumBins = Defaults.NumBins;
         }
 
-        internal static string RegistrationName = "MutualInformationFeatureSelectionTransform";
-
-        /// <summary>
-        /// A helper method to create <see cref="IDataTransform"/> for selecting the top k slots ordered by their mutual information.
-        /// </summary>
-        /// <param name="env">Host Environment.</param>
-        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
-        /// <param name="labelColumn">Column to use for labels.</param>
-        /// <param name="slotsInOutput">The maximum number of slots to preserve in output.</param>
-        /// <param name="numBins">Max number of bins for R4/R8 columns, power of 2 recommended.</param>
-        /// <param name="columns">Columns to use for feature selection.</param>
-        public static IDataTransform Create(IHostEnvironment env,
-            IDataView input,
+        private IHost _host;
+        private readonly (string input, string output)[] _columns;
+        private readonly string _labelColumn;
+        private readonly int _slotsInOutput;
+        private readonly int _numBins;
```

Comment:
How hard would it be to convert it into ColumnInfo with column specific params?

Reply:
Currently, we specify many input columns and we keep the top k slots across all input columns based on mutual info. We could add a ColumnInfo that allows to do multiple of these transformations at the same, though I am not sure why one would do it in the first place. It would not complicate the code too much, just wrap everything we have in the Fit method around a for loop that replicates the same actions on different columninfo elements.

---
In reply to: [235535904](https://github.com/dotnet/machinelearning/pull/1683#discussion_r235535904) [](ancestors = 235535904)

Reply:
Do you think it would be beneficial to add a columnInfo?

---
In reply to: [235541509](https://github.com/dotnet/machinelearning/pull/1683#discussion_r235541509) [](ancestors = 235541509,235535904)

Reply:
With columnInfo you can remove restraint from reconciler to have only one column to process, but in same time i'm not sure how much useful it is.

---
In reply to: [236792186](https://github.com/dotnet/machinelearning/pull/1683#discussion_r236792186) [](ancestors = 236792186,235541509,235535904)